adr/ADR-43.md

```diff
@@ -22,19 +22,28 @@ Related issues [#3268](https://github.com/nats-io/nats-server/issues/3268)
 
 ## Per-Message TTL
 
-We will allow a message to supply a TTL using a header called `Nats-TTL` followed by the duration as seconds.
+### General Behavior
+
+We will allow a message to supply a TTL using a header called `Nats-TTL` followed by the duration as seconds or as a Go duration string like `1h`.
 
 The duration will be used by the server to calculate the deadline for removing the message based on its Stream 
 timestamp and the stated duration.
 
-The TTL may not exceed the Stream MaxAge. The shortest allowed TTL would be 1 second. When no specific TTL is given
-the MaxAge will apply.
-
-Setting the header `Nats-No-Expire` to `1` will result in a message that will never be expired.
+Setting the header `Nats-TTL` to `never` will result in a message that will never be expired.
 
 A TTL of zero will be ignored, any other unparsable value will result in a error reported in the Pub Ack and the message
 being discarded.
 
+When a message with the `Nats-TTL` header is published to a stream with the feature disabled the message will be rejected with an error.
+
+### Sources and Mirrors
+
+When messages arrive over a Source or a Mirror processing is a little bit different: Where for a normal client-published message we would reject it if the stream lacks the feature from a Source/Mirror we would accept it.
+
```

Comment:
Would be good to describe the behaviour more explicitly:
* what will happen if the message is accepted from Source/Mirror but feature is disabled
* what happens if feature is enabled.

I know it's obvious for us, but might not be for other engineers.

Reply:
```....however if the Source or Mirror has the `AllowMsgTTL` set those message with the header will be processed as above.```

It's identical to the sections above? Doesn't seem useful to repeat it and then later forget to update it here

Reply:
I agree, we should not repeat anything, but I would love bit more clarity:
- "Above" is bit ambiguous - maybe mention that we mean behavior mentioned in `General Behavior` section.
- "if the stream lacks the feature from a Source/Mirror we would accept it." - I would explicitly add that we would accept it, but not process the msg TTL.

I had to read that section twice while understanding the topic. Just want to avoid any confusion for readers :).

Reply:
rewrote the section, how's the new one?

Reply:
looks much cleaner!  LGTM